### DISCUSSION TITLE
Fontawesome & markdown

### DISCUSSION THREAD

**Discussion by celinehaddad** [FortAwesome/Font-Awesome#19627] [Q&A]
Hey all,

I am trying to add Fontawesome support to a Next.js TypeScript app I am working on that uses ContentLayer to manage markdown files. I want to be able to add `:fontawesome-solid-check:` into a markdown file, and have my app convert that to the html version & render the correct icon. I don't want to host the library myself for performance reasons. Any ideas or resources on how to get this set up?

**Comment by tagliala** [comment]
Hi,

> I don't want to host the library myself for performance reasons

Maybe the `CSS` implementation hosted at https://cdnjs.com/libraries/font-awesome could be useful for this use case?

Note: `cdnjs` is a third party CDN like others

**Comment by celinehaddad** [comment]
@tagliala I created a kit and added it to the my app, but I want to be able to add `:fa-heart:` to my markdown file and have my app convert it to render the corresponding icon. I don't think I will need the third party CDN to do this, but not sure

**Reply by celinehaddad** [reply]
I am also using the remark-emoji plugin to convert regular emojis. Just want to be able to convert fontawesome icons too

**Reply by tagliala** [reply]
Sorry, I probably did not understand the opening post

The purpose is to convert `:fa-heart:` to `<span class="fa-solid fa-heart"></span>` and you are looking for a markdown plugin to do that?

**Reply by celinehaddad** [reply]
No worries, yes exactly :)

**Comment by jhoney124** [comment]
yes exactly
